jsonschema-validation.xml

```diff
@@ -539,26 +554,60 @@
 
             <section title="Implementation Requirements">
                 <t>
-                    The "format" keyword functions as both an annotation
-                    and as an assertion.  While no special effort is required to
-                    implement it as an annotation conveying semantic meaning, implementing
-                    validation is non-trivial.
-                </t>
-                <t>
-                    Implementations MAY support the "format" keyword as a validation assertion.
-                    Should they choose to do so:
-
+                    The "format" keyword functions as an annotation, and optionally as an assertion.
+                    Declaring the format vocabulary in "$vocabulary" with a value of true indicates
+                    that an implementation MUST treat the keyword as an assertion.
+                    This means that:
                     <list>
                         <t>they SHOULD implement validation for attributes defined below;</t>
                         <t>they SHOULD offer an option to disable validation for this keyword.</t>
                     </list>
+                </t>
 
+                <t>
+                    Due to the complexity involved in fully validating some format attributes
+                    defined in this specification, implementations MAY provide only limited
+                    validation support for some format attributes.  Implementations SHOULD
+                    document any such intentional limitations.
```

Comment:
I hear you on this.
One of the reasons I was going to shunt this issue to draft-9 was I wanted to provide a MINIMUM requirement for the MUST defined above in line 559.

Otherwise, the MUST on 559 holds no actual requirement, because it's then defined that meeting said MUST is actually meeting two SHOULD requirements (and we're back to... whatever the implementation wants to do).

The point of the issues around format are providing AT LEAST SOMEthing when format is used if required.

Reply:
This does require at least something- it just does not define what that something is, and there is NO WAY IN HELL we will get any consensus from implementors on defining a minimum for each format attribute within the next couple weeks.

What this does is ensures that the keyword produces a validation result, and gives both the implementation and schema author _some_ level of control.  Previously, the schema author had _no_ control over how things were processed.

This does not solve everything but it's a big improvement and I'm not willing to let the perfect be the enemy of the good here.  For me personally, `format` is unsalvageable as any sort of interoperable feature and should be discouraged (if not formally deprecated or removed- it's far too heavily used for that right now) in favor of new vocabularies.

Reply:
@Relequestual OK now that I'm more awake and have had a chance to think about it more (I really shouldn't- SHOULD NOT? - respond to comments right when I first wake up) I think that having the vocabulary true/false option means that we can strengthen these requirements without having a huge fight over the minimal requirements.  Which can be pretty minimal with a warning that they may be strengthened in the future.

We can do this because the default usage of the vocabulary is still `false`, which is essentially the same as what it has always been.  As is noted in the text, even when the vocabulary is `false`, an implementation MAY try to validate it anyway.

So let's do this:

* If the vocabulary is `true`, then all of the format attributes MUST be supported to some minimum level along the lines of #54.  Otherwise the implementation must raise an unsupported vocabulary error.
* If the vocabulary is `false`, then it's all MAY.  Which includes the SHOULD level that is currently specified (and, as far as I can tell, very frequently ignored to some degree or another).

Again, since the default is `false`, this is an entirely backwards-compatible change.

I will revise the PR.  I think I may move a bit of this text and quite a bit more commentary into an appendix of guidance on implementing and (if necessary e.g. because OpenAPI has a huge format registry already) extending`format`.

Reply:
I think this is a totally reasonable and proportionate way forward.
I'm happy with a super bare minimum level of required validation in the instance where the format vocabulary is `true`: for example, email matches `^.+@.+\..+$`. Something that catches the equivilence of a glancing "huh... well that's clearly wrong" type issue.

Ultimatly, JSON Schema is (normally) first pass on validation, apart from the instances where it's baked into a database or the like.